sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -819,8 +819,10 @@ object JdbcUtils extends Logging {
     if (null != customSchema && customSchema.nonEmpty) {
```

Comment:
Oh, I see. We need to accept a nested schema in `customSchema`? I checked the original PR https://github.com/apache/spark/pull/18266, but I couldn't find test cases for nested schemas. So, I'm not sure this is an expected behaviour... cc: @ wangyum

Reply:
I don't know which JDBC server supports nested schema. But IIUC this feature is to specify the type, and I think it can be used to specify the data type of nested fields as well.

Reply:
Yea, it can be and accepting nested fields looks okay. Either way, I think we need more test cases for `customeSchema` with nested fields, arrays, map, ...

Reply:
JDBC spec mentions the STRUCT type, for example https://docs.oracle.com/javase/8/docs/api/java/sql/Types.html#STRUCT.

At least, you can access to Spark cluster from another Spark cluster via JDBC ;-)